charts/apisix/templates/clusterrole.yaml

```diff
@@ -0,0 +1,42 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{- $kubernetesTokenFile := "/var/run/secrets/kubernetes.io/serviceaccount/token" -}}
+{{- $create := false -}}
+{{- if .Values.discovery.enabled -}}
+{{-   with .Values.discovery.registry -}}
+{{-     if hasKey . "kubernetes" -}}
+{{-       if .kubernetes.client -}}
+{{-         if eq .kubernetes.client.token_file $kubernetesTokenFile -}}
+{{-           $create = true -}}
```

Comment:
If a custom token path is specified and it is not the kubernetes token path, the helm chart simply will not create the cluster role and binding.

Reply:
OK, but how can this user mount the token to the container?

Reply:
Actually, my idea is to provide the flexibility for users to decide whether or not to use the kubernetes service account token file. In case users want to use it, we generate the RBAC. Otherwise, users might want to mount other tokens using e.g. Secret to a specific path, in which the RBAC is not needed to be generated. In this case, users will need to configure the custom tokens to be used in prior.

Reply:
Oh I think I got what you mean, there is currently no way to mount the token.

Reply:
I thought for a while, and I think maybe a custom token path is not useful when users deploy Apache APISIX on Kubernetes, what they want is a custom token, the path is not critical. So as long as users can mount the custom token, the purpose can be achieved. 

So what we can do is support mount a custom serviceaccount. This PR is large enough, l just add another issue https://github.com/apache/apisix-helm-chart/issues/322 to track it.

Reply:
I'll have to agree with you, sounds good to me. Thanks